### ISSUE TITLE
1.0.0 version needs write access to m2

### ISSUE THREAD

**Issue by Andre0991** [xsc/ancient-clj#14]
Hi.

My company uses `ancient-clj` for getting the latest version of some dependencies. Those deps are store in our private repository, in an s3 bucket.

The thing is that starting with version 1.0.0, it seems that `ancient-clj` is trying to write to `~/.m2`, which is not possible in our env, as we're running the code in an AWS lambda and it only has write access to `/tmp`.

So the lambda get errors such as:

```
2021-06-01T19:54:16.941Z <request-id-not-set-by-lambda-runtime> WARN  Failed to create parent directories for tracking file /home/sbx_user1051/.m2/repository/<some-repo-name>/<some-repo-name>/resolver-status.properties
```

We haven't investigated this issue too deeply, but downgrading `ancient-clj` seems to solve it.

Some questions:
- Is this assumption correct? Did 1.0.0 start writing to m2 for getting the latest version of some dep?
- If so, is there any way of avoiding that? If the assumption is wrong, sorry for the noise.

**Comment by xsc** [comment]
@Andre0991 You're absolutely right. With version 1.0.0, `ancient-clj` switched to the same dependency resolution mechanism that Leiningen uses (via `pomegranate`) to be able to easily use existing wagons and other configurations. A side-effect of that seems to be that it (a) includes versions from the local repository (even if they were locally installed and don't exist remotely) and (b) always tries to update the metadata in `~/.m2/` when doing a version range request.

The location of the local repo is configurable via the options map that can be passed to the ancient-clj loader, e.g.:

```clojure
(def loader (ancient/loader {:local-repo "/tmp/ancient-repo"}))
(ancient/latest-version loader 'ancient-clj)
;; => {:version-clj.core/version {:version [(1 0 0)], :qualifiers #{}, :snapshot? false, :qualified? false},
;;     :ancient-clj.artifact/version "1.0.0"}
```

This might already solve your problem. I did not find a way to avoid the write access completely, but maybe we should just rely on a temporary folder by default.

**Comment by Andre0991** [comment]
Thanks!  We'll use `loader`, yes.

**Comment by Andre0991** [comment]
Hi @xsc.

I've tried to use this custom loader, but it seems that the latest released version (1.0.0) does not have `loader` and `latest-version`:

```
(clojure.repl/dir 'ancient)
artifact-outdated-string?
artifact-outdated?
default-repositories
latest-version!
latest-version-string!
maybe-create-loader
maybe-create-loaders
read-artifact
version-strings!
versions!
versions-per-repository!
nil
```

This is `core.clj` in 1.0.0: https://github.com/xsc/ancient-clj/blob/3cdf6a99ef6cdd19ec2ec6018611079cc4400d22/src/ancient_clj/core.clj

Those vars were introduced in SNAPSHOT versions.

Am I missing something? Or should `ancient-clj` get a release?

**Comment by xsc** [comment]
@Andre0991 You're not missing anything - I did indeed forget to release the most recent changes. (Probably being cautious since I wanted to verify that it works with lein-ancient first.) Now you can use:

```clj
[ancient-clj "2.0.0"]
```
